subprojects/core/src/main/java/org/gradle/plugin/software/internal/SoftwareTypeRegistry.java

```diff
@@ -34,4 +34,9 @@ public interface SoftwareTypeRegistry {
      * Returns a set of the software types available along with their model types and associated plugins.
      */
     Set<SoftwareTypeImplementation> getSoftwareTypeImplementations();
+
```

Comment:
💭 It should be noted on `SoftwareTypeRegistry#getSoftwareTypeImplementations()` that calling this method should be done only after all plugin classes are registered, and that it is an error to register afterwards.

A test in `DefaultSoftwareTypeRegistryTest` that tries this and demos the failure would be a good addition, too.

Reply:
Good point.  Done.